src/ts/index.ts

```diff
@@ -316,22 +310,21 @@ export function hashOrder(domainSeparator: string, order: Order): string {
 /**
  * Returns the signature for the specified order.
  * @param owner The owner for the order used to sign.
- * @param domainSeparator The domain separator to add to the digest used for
- * signing. See {@link hashOrder} for more details.
+ * @param domainSeparator The domain separator to add to the digest. This is
+ * used by the smart contract to ensure order's can't be replayed across
+ * different applications (domains), but also different deployments (as the
+ * contract chain ID and address is used to salt the domain separator value).
  * @param order The order to sign.
- * @return Signature for the order.
+ * @return Hex-encoded signature for the order.
  */
 export function signOrder(
-  owner: Wallet,
+  owner: Signer,
   domainSeparator: string,
   order: Order,
-): Signature {
-  const digest = hashOrder(domainSeparator, order);
-  // NOTE: We need to sign directly with the signing key here on an Ethers.js
-  // wallet and not the `Signer` API, as the latter uses the `"\x19Ethereum
-  // Signed Message:\n" + len(message)` prefix. This is likely to change in the
-  // future as wallets generally don't provide a method for doing this as it is
-  // considered unsafe.
-  // <https://github.com/gnosis/oba-contracts/issues/129>
-  return owner._signingKey().signDigest(digest);
+): Promise<string> {
+  return owner.signMessage(
```

Comment:
This will allow frontends to sign orders, right? 🎉

Reply:
Correct!